projects/ad469x_iio/app/app_config_stm32.c

```diff
@@ -0,0 +1,352 @@
+/**
+ * @brief 	Initialize the STM32 system peripherals
+ * @return	None
+ */
+void stm32_system_init(void)
+{
+	HAL_Init();
+	SystemClock_Config();
+#if (INTERFACE_MODE == SPI_DMA)
+	/* If SPI_DMA mode is enabled, we don't need DMA initialization
+	 * and Timer 2 initialization which is used for generating the
+	 * chip select signals.
+	 */
+	MX_DMA_Init();
+	MX_TIM2_Init();
+	MX_TIM8_Init();
+#endif
+	MX_GPIO_Init();
+	MX_UART5_Init();
+	MX_I2C1_Init();
+	MX_TIM1_Init();
+	MX_SPI1_Init();
+#if (INTERFACE_MODE == SPI_INTERRUPT)
+	HAL_NVIC_DisableIRQ(DMA2_Stream0_IRQn);
+#else
+	HAL_NVIC_DisableIRQ(EXTI9_5_IRQn);
+#endif
+}
+
+/**
+ * @brief 	Starts the timer signal generation for
+ *          PWM and OC channels all at once.
+ * @return	None
+ */
+void stm32_timer_enable(void)
+{
+#if (INTERFACE_MODE == SPI_DMA)
+	sdesc = p_ad469x_dev->spi_desc->extra;
+
+	no_os_pwm_enable(sdesc->pwm_desc); // CS Pwm;
+	no_os_pwm_enable(pwm_desc); // CNV PWM
+#endif
+}
+
+/**
+ * @brief Disable PWM signals
+ * @return None
+ */
+void stm32_timer_stop(void)
+{
+#if (INTERFACE_MODE == SPI_DMA)
+	int ret;
+	sdesc = p_ad469x_dev->spi_desc->extra;
+
+	no_os_pwm_disable(pwm_desc);// CNV PWM
+	no_os_pwm_disable(sdesc->pwm_desc); // CS PWM
+
+	TIM8->DIER &= ~TIM_DIER_CC1DE;
+
+	/* Disable RX DMA */
+	CLEAR_BIT(sdesc->hspi.Instance->CR2, SPI_CR2_RXDMAEN);
+#endif
+}
+
+/**
+ * @brief  Abort DMA Transfers
+ * @return None
+ */
+void stm32_abort_dma_transfer(void)
+{
+#if (INTERFACE_MODE == SPI_DMA)
+	int ret;
+
+	sdesc = p_ad469x_dev->spi_desc->extra;
+
+	ret = no_os_dma_xfer_abort(sdesc->dma_desc, sdesc->rxdma_ch);
+	if (ret) {
+		return ret;
+	}
+
+	ret = no_os_dma_xfer_abort(sdesc->dma_desc, sdesc->txdma_ch);
+	if (ret) {
+		return ret;
+	}
+#endif
+}
+
+/**
+ * @brief  Configures the chip select pin as output mode.
+ * @param  is_gpio[in] Mode of the Pin
+ * @return	None
+ */
+void stm32_cs_output_gpio_config(bool is_gpio)
+{
+#if (INTERFACE_MODE == SPI_DMA)
+	struct no_os_gpio_desc* cs_gpio_desc;
+
+	if (is_gpio) {
+		cs_pwm_gpio_params.extra = &stm32_cs_gpio_extra_init_params;
+	} else {
+		cs_pwm_gpio_params.extra = &stm32_cs_pwm_gpio_extra_init_params;
+	}
+
+	no_os_gpio_get(&cs_gpio_desc, &cs_pwm_gpio_params);
+#endif
+}
+
+
+/**
+ * @brief 	Configures the conversion pin as output mode.
+ * @param is_gpio[in] Mode of the Pin
+ * @return	None
+ */
+void stm32_cnv_output_gpio_config(bool is_gpio)
+{
+#if (INTERFACE_MODE == SPI_DMA)
+	struct no_os_gpio_desc* cnv_gpio_desc;
+
+	if (is_gpio) {
+		pwm_gpio_params.extra = &stm32_gpio_cnv_extra_init_params;
+
+	} else {
+		pwm_gpio_params.extra = &stm32_pwm_gpio_extra_init_params;
+	}
+	no_os_gpio_get(&cnv_gpio_desc, &pwm_gpio_params);
+#endif
+}
+
+void tim1_config(void)
+{
+#if (INTERFACE_MODE == SPI_DMA)
+	TIM1->EGR = TIM_EGR_UG;// Generate update event
+	TIM1->CR2 =  (TIM_CR2_MMS_COMPARE_PULSE *
+		      TIM_CR2_MMS_0);
+#endif
+}
+
+void tim8_config(void)
```

Comment:
these register mainpulations should be on drivers/platform/stm32/*?

Reply:
A few bitfield configurations aren't really handled in the platform driver layer. Also, since the DMA implementation is timing critical, we would want to keep it as in direct reg acces mode